ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,281 @@
 #include <catch2/catch.hpp>
 
 #include "azure_connection.h"
+#include <fstream>
 
-TEST_CASE("testing class azure_connection", "azure_connection") {}
+TEST_CASE("Testing Azure Connection Class", "azure-connection")
+{
+
+    Azure::Storage::Blobs::BlobContainerClient azure_client =
+      Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
+        std::getenv("AZURE_STORAGE_CONNECTION_STRING"), "myblobcontainer1");
+    bool exists = false;
+
+    const std::string object_name = "test_object";
+    const std::string file_name = "test_object.txt";
+    const std::string path = "./" + file_name;
+    const std::string object_name_2 = "test_object_2";
+    const std::string file_name_2 = "test_object_2.txt";
+    const std::string path_2 = "./" + file_name_2;
+    const std::string non_exi_object_key = "test_non_exist";
+    const std::string non_exi_file_name = "test_non_exist.txt";
+
+    std::ofstream file(file_name);
+    const std::string payload = "Test payload";
+    const std::string payload_substr = "payload";
+    file << payload;
+    file.close();
+
+    std::ofstream file_2(file_name_2);
+    const std::string payload_2 = "Testing offset and substring";
+    const std::string payload_2_substr = "offset";
+    file_2 << payload_2;
+    file_2.close();
+
+    SECTION(
+      "(Checks if connection to a non-existing bucket fails gracefully).", "[azure-connection]")
+    {
+        REQUIRE_THROWS_WITH(azure_connection("Bad_bucket", ""), "Bad_bucket : No such bucket.");
+    }
+
+    SECTION("(Check if an object exists in Azure)", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "obj_exists_");
+
+        // Object does not exist yet so there should be 0 objects.
+        CHECK(conn.object_exists(file_name, exists) == 0);
+        CHECK(exists == false);
+
+        auto blob_client = azure_client.GetBlockBlobClient("obj_exists_" + file_name);
+        blob_client.UploadFrom(file_name);
+
+        // Object exists now in the container.
+        CHECK(conn.object_exists(file_name, exists) == 0);
+        CHECK(exists == true);
+
+        blob_client.Delete();
+    }
+
+    SECTION("(Lists Azure objects under the text bucket).", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "list_obj_");
+        std::vector<std::string> objects;
+
+        // Total objects to insert in the test.
+        const int32_t num_objects = 11;
+        // Prefix for objects in this test.
+        const std::string prefix = "list_obj_";
+        const bool list_single = true;
+
+        // No matching objects. Object size should be 0.
+        CHECK(conn.list_objects(prefix + file_name, objects, !list_single) == 0);
```

Comment:
I would create another variable for prefix + file_name.

Reply:
Taken onboard https://github.com/wiredtiger/wiredtiger/pull/8747/commits/b8979db4be25c1b70cd0fc6ceba12047b1b27c25